temboardui/plugins/monitoring/__init__.py

```diff
@@ -410,40 +409,68 @@ def collector(app, address, port, key):
                          "Please consider upgrading to version 5 at least.")
         else:
             logger.exception(str(e))
-        # Update collector status
-        update_collector_status(
-            worker_session,
-            instance_id,
-            u'FAIL',
-            last_pull=datetime.utcnow(),
-        )
-        worker_session.commit()
+            # Update collector status
+            update_collector_status(
+                worker_session,
+                instance_id,
+                u'FAIL',
+                last_pull=datetime.utcnow(),
+            )
+            worker_session.commit()
```

Comment:
We don't want to change collector status if /monitoring/history returns 404 error because it means the agent is in a lower version than 5, meaning that only the push mode is available for this one.